WcaOnRails/app/webpacker/components/UserBadge.js

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import {
+  Button, Icon, Label, Popup,
+} from 'semantic-ui-react';
+import UserAvatar from './UserAvatar';
+import I18n from '../lib/i18n';
+
+import '../stylesheets/user_badge.scss';
+
+/**
+ * @param {{ leader: boolean, senior_member: boolean}} user
+ * @returns {string[]}
+ */
+export function subtextForMember(user) {
+  if (user.leader) {
+    return [I18n.t('about.structure.leader')];
+  }
+  if (user.senior_member) {
+    return [I18n.t('about.structure.senior_member')];
+  }
+
+  return [];
+}
+
+/**
+ * @param {{ teams: { friendly_id: string, leader: boolean }[]}} user
+ * @param {{ friendly_id: string, name: string }[]} officerTitles
+ * @returns {string[]}
+ */
+export function subtextForOfficer(user, officerTitles) {
+  const positions = user.teams
+    .map((team) => {
+      const title = officerTitles.find((t) => t.friendly_id === team.friendly_id);
+      if (title) return title.name;
+      return '';
+    })
+    .filter(Boolean);
+
+  if (user.teams.filter((team) => team.friendly_id === 'wfc' && team.leader).length > 0) {
+    positions.push(I18n.t('about.structure.treasurer.name'));
+  }
+
+  return positions;
+}
+
+function UserBadge({
+  user, subtexts = [], background = '', badgeClasses = '', senior = false, leader = false,
+}) {
+  let classes = `user-badge ${badgeClasses}`;
```

Comment:
bringing on the [classnames](https://www.npmjs.com/package/classnames) package should improve the readability of this code if this class isn't already included.

Reply:
Ah that sounds good, I hadn't heard of that